core/GlobalState.cc

```diff
@@ -1536,6 +1536,8 @@ bool GlobalState::tryCommitEpoch(u4 epoch, bool isCancelable, function<void()> t
     // Typechecking does not run under the mutex, as it would prevent another thread from running `tryCancelSlowPath`
     // during typechecking.
     typecheck();
```

Comment:
Context for Mick: This method is called whenever we run the slow path. It returns 'true' if the slow path completed successfully, or 'false' if it was canceled. We pass it a lambda (typecheck) that runs the slow path.